crates/dbsp/src/trace/ord/indexed_zset_batch.rs

```diff
@@ -390,6 +396,53 @@ where
     }
 }
 
+impl<K, V, R, O> Encode for OrdIndexedZSet<K, V, R, O>
+where
+    K: DBData,
+    V: DBData,
+    R: DBWeight,
+    O: OrdOffset,
+{
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        let len: usize = self.len();
+        Encode::encode(&len, encoder)?;
+        let mut n = 0;
+        let mut cursor = self.cursor();
+        while cursor.key_valid() {
+            while cursor.val_valid() {
+                Encode::encode(cursor.key(), encoder)?;
+                Encode::encode(cursor.val(), encoder)?;
+                Encode::encode(&cursor.weight(), encoder)?;
+                n += 1;
+                cursor.step_val();
+            }
+            cursor.step_key();
+        }
+        assert_eq!(n, len);
```

Comment:
should this be debug_assert?

Reply:
does the transport have some kind of checksums?

Reply:
> should this be debug_assert?

Changed.

> does the transport have some kind of checksums?

No.